wrapper/src/test/java/integration/util/AuroraTestUtility.java

```diff
@@ -626,6 +630,33 @@ protected void makeSureInstancesUp(List<String> instances, boolean finalCheck)
     }
   }
 
+  // Attempt to run a query after the instance is down.
+  // This should initiate the driver failover, first query after a failover
+  // should always throw with the expected error message.
+  public void assertFirstQueryThrows(Connection connection, String expectedSQLErrorCode) {
+    final SQLException exception =
+        assertThrows(
+            SQLException.class,
+            () ->
+                queryInstanceId(
+                    TestEnvironment.getCurrent().getInfo().getRequest().getDatabaseEngine(),
+                    connection));
+    assertEquals(expectedSQLErrorCode, exception.getSQLState());
+  }
+
+  protected void assertFirstQueryThrows(Statement stmt, String expectedSQLErrorCode) {
+    final SQLException exception =
+        assertThrows(
+            SQLException.class,
+            () ->
+                executeInstanceIdQuery(
+                    TestEnvironment.getCurrent().getInfo().getRequest().getDatabaseEngine(), stmt));
+    assertEquals(
+        expectedSQLErrorCode,
+        exception.getSQLState(),
+        "Unexpected SQL Exception: " + exception.getMessage());
+  }
```

Comment:
Can we pass in the expected SQLException instead of the expectedSQLErrorCode? For instance, FailoverSucceededException, then just assert for that exception.
```
assertThrows(
            FailoverSucceededException.class,
            () -> queryInstanceId(...);
```
This way we won't have to check the SQL state.